File: PracticasProgramacion/Practica_01/cpp/HolaMundo.cpp
/**
 * Practica_01 - HolaMundo.cpp
 * Implementación de la clase HolaMundo
 */

#include "HolaMundo.h"
#include <iostream>

HolaMundo::HolaMundo() : mensaje_("Hola mundo") {}

HolaMundo::HolaMundo(const std::string& mensaje) : mensaje_(mensaje) {}

void HolaMundo::mostrar() const {
    std::cout << mensaje_ << std::endl;
}

void HolaMundo::setMensaje(const std::string& mensaje) {
    mensaje_ = mensaje;
}

std::string HolaMundo::getMensaje() const {
    return mensaje_;
}

File: PracticasProgramacion/Practica_01/python/hola_mundo.py
"""
Clase HolaMundo: encapsula un mensaje y su visualización.
Atributo mensaje_ privado; constructores default y parametrizado.
"""


class HolaMundo:
    def __init__(self, mensaje: str = "Hola mundo") -> None:
        self._mensaje = mensaje

    def mostrar(self) -> None:
        print(self._mensaje)

    def set_mensaje(self, mensaje: str) -> None:
        self._mensaje = mensaje

    def get_mensaje(self) -> str:
        return self._mensaje
